### ISSUE TITLE
Site Editor: Silent tab stop and focus loss when tabbing from navigation to canvas iframe

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51540]
### Description

In the Site Editor, when tabbing from the navigation panel to the editor iframe, there's a silent tab stop. Silent tab stops should be avoided at all costs, as they're a very confusing experience for all keyboard users: after pressing the Tab key, nothing seems to happen. No visual feedback on where focus is. No audible feedback for screen reader users as well.

It took me a while to debug. Turns out:

When the Site editor loads, the editor canvas is in 'view' mode. In this state, it shows the navigation panel and the editor iframe with the site preview. Within the iframe, the block editor is in 'edit' mode. As such, `useTabNav` renders two fodusable div elements, before and after the iframe.

These two fosuable dive elements are in the tab sequence. The first one is right after the navigation panel. The second one is right after the iframe.

When tabbing through the UI, the focusable div between the navigation and the iframe is a silent tab stop.

<img width="1587" alt="Screenshot 2023-06-14 at 11 56 56" src="https://github.com/WordPress/gutenberg/assets/1682452/359a5875-1033-4e74-8565-8b4955238674">

Additionally, when the div element receives focus, the `useTabNav` behavior kicks in and removes the `tabindex="0"` attribute from the div.

<img width="1675" alt="Screenshot 2023-06-14 at 11 57 26" src="https://github.com/WordPress/gutenberg/assets/1682452/229fd230-a484-47c9-9482-71d3540fbeb5">

At this point, there's a focus loss: the element that had focus is no longer focusable. Focus fallbacks to the body element. See the screenshot below, where I'm using a small debugging custom plugin to dump to the console the current activeElement.

<img width="1554" alt="Screenshot 2023-06-14 at 12 16 06" src="https://github.com/WordPress/gutenberg/assets/1682452/45bdefd3-9d7c-49a6-a0d4-95fafe2d7ba6">

At this point, when pressing the Tab key agaiin, the iframe receives focus. This happens only because of navite browsers behavior. In fact, after a focus loss, modern browsers attempt to restart the tab sequence from the closes place where focus previously was. For more details on this browsers native feature, see for example [sequential focus navigation starting point](https://bugs.chromium.org/p/chromium/issues/detail?id=454172) for Chrome. This browsers behavior isn't guaranteed to always work and we can't rely on it.

Regardless, there shouldn't be silent tab stops. Also, focus losses should be avoided at all costs.

### Step-by-step reproduction instructions

- Go to teh site editor, stay in view mode.
- Tab through the UI.
- Observe there's a silent tab stop between the navigation panel and the iframe.
- Once focus is on the iframe Tab backwards and then forwards agaiin.
- Observe the silent tab stop is gone.
- Switch the editor canvas to edit mode.
- Switch back the editor canvas to view mode.
- Navigate again through the ui with the Tab key.
- Observe the silent tab stop is there again.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
